plugins/backstage-plugin-coder/src/components/CoderWorkspacesCard/WorkspacesListItem.tsx

```diff
@@ -226,6 +230,37 @@ export const WorkspacesListItem = ({
   );
 };
 
+const deletingStatuses: readonly WorkspaceStatus[] = ['deleting', 'deleted'];
+const offlineStatuses: readonly WorkspaceStatus[] = [
+  'stopped',
+  'stopping',
+  'pending',
+];
+
+type AvailabilityStatus = 'online' | 'offline' | 'deleting';
+
+function getAvailabilityStatus(workspace: Workspace): AvailabilityStatus {
+  const currentStatus = workspace.latest_build.status;
+
+  // When a workspace is being deleted, there is a good chance that the agents
+  // will still show as connected/connecting. If this check isn't done before
+  // looking at the agent statuses, a deleting workspace might show up as online
+  if (deletingStatuses.includes(currentStatus)) {
+    return 'deleting';
+  }
+
+  if (offlineStatuses.includes(currentStatus)) {
+    return 'offline';
+  }
+
+  const agentStatuses = getWorkspaceAgentStatuses(workspace);
+  const showAsAvailable = agentStatuses.every(
+    status => status === 'connected' || status === 'connecting',
```

Comment:
`connecting` does not indicate online yet, I think, and `connected` might not either, depends on the lifecycle state. There are other states that indicate whether the agent is ready to be used.  It can also depend on whether `login_before_ready` is set.

Here is a full example of every state possible and whether it is connectable or not, maybe we can make something similar: https://github.com/coder/jetbrains-coder/blob/3096448efc97f262c86235da20b8729b0c8b5f8e/src/main/kotlin/com/coder/gateway/models/WorkspaceAndAgentStatus.kt

Reply:
Update to say that we were talking about this, and I forgot that all the Coder plugin does is link to the dashboard so it is not really critical that we check the states so thoroughly.  But no harm in it either!

Reply:
Yeah, I think you're right – we will want something like this in the codebase soon.
Since the workspaces for the plugin don't have access to all the properties just yet, I'm going to use some stopgap logic to account for agents a little more, but it won't be super thorough

Next week, though, when I focus on adding support for the Coder API, that's when I'll try to revamp all this, and I'll be using the JetBrains code as my main example